services/src/main/java/org/keycloak/organization/validator/OrganizationMemberValidator.java

```diff
@@ -135,7 +136,9 @@ private static Set<String> resolveExpectedDomainsForManagedUser(ValidationContex
 
         for (IdentityProviderModel broker : brokers) {
             String domain = broker.getConfig().get(OrganizationModel.ORGANIZATION_DOMAIN_ATTRIBUTE);
-
+            if (ANY_DOMAIN.equalsIgnoreCase(domain)) {
+                organization.getDomains().map(OrganizationDomainModel::getName).forEach(domains::add);
```

Comment:
nit: Maybe continue here, otherwise `ANY` is added to the domains set?

Reply:
Ahhh, yeah, good catch!